### PR TITLE
feat: Support WellKnownSystemCerts in BackendTLSPolicy

### LINKED ISSUE #2775: Support Well Known System Certs in BackendTLSPolicy
*Description*:
>Describe the issue.

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

### PR DESCRIPTION
**Pull request by liorokman**
**What this PR does / why we need it**:
Use the system trust store from the Envoy Proxy image if asked to do so in BackendTLSPolicy.

**Which issue(s) this PR fixes**:
Fixes #2775

### CODE REVIEW

File: internal/xds/translator/translator.go [hunk 1]
```diff
@@ -613,17 +618,42 @@ func buildXdsUpstreamTLSCASecret(tlsConfig *ir.TLSUpstreamConfig) *tlsv3.Secret
...
+
+	if tlsConfig.UseSystemTrustStore {
+		tlsCtx = &tlsv3.UpstreamTlsContext{
+			CommonTlsContext: &tlsv3.CommonTlsContext{
+				ValidationContextType: &tlsv3.CommonTlsContext_ValidationContext{
+					ValidationContext: &tlsv3.CertificateValidationContext{
+						TrustedCa: &corev3.DataSource{
+							Specifier: &corev3.DataSource_Filename{
+								// This is the default location for the system trust store
+								// on Debian derivatives like the envoy-proxy image being used by the infrastructure
```
Comment (guydc) [inline internal/xds/translator/translator.go:631]: Is that also the correct path for distroless? I think that all EG releases are now based on distroless.

Comment (liorokman) [reply internal/xds/translator/translator.go:631]: Yes, `distroless` is a Debian derivative.  See [here](https://github.com/GoogleContainerTools/distroless?tab=readme-ov-file#how-do-i-use-distroless-images).

**Review by guydc — APPROVED** [review]
LGTM, thanks!

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Pull request merged**